client/recordUtil.js

```diff
@@ -250,25 +250,40 @@ const mergeRecord = (record1, record2) => {
 
 /**
  * Within an array of [record, object], merge items whose records have the
- * same objectId.
+ * same objectId and same actions.
+ * Requirements:
+ * 1) should not combine any CREATE
+ * 2) should not combine any DELETE
+ * 3) all of UPDATE should be combined to single update
  * @param {Array} recordsAndObjects Same format as input to resolveRecords().
  * @returns {Array}
  */
 const mergeRecords = (recordsAndObjects) => {
-  const objectIdMap = {} // map of objectId to [record, object]
+  var mergedResult = []
   recordsAndObjects.forEach((recordAndObject) => {
     const record = recordAndObject[0]
-    const object = recordAndObject[1]
     const id = JSON.stringify(record.objectId)
-    if (objectIdMap[id]) {
-      const mergedRecord = mergeRecord(objectIdMap[id][0], record)
-      objectIdMap[id] = [mergedRecord, object]
-    } else {
-      objectIdMap[id] = recordAndObject
+
+    if (!mergedResult.length ||
+        record.action === proto.actions.CREATE ||
+        record.action === proto.actions.DELETE) {
+      mergedResult.push(recordAndObject)
+      return
```

Comment:
what is `return` doing here? It doesn't stop or break forEach
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#Description